tests/Unit/Resource/Detectors/K8sTest.php

```diff
@@ -190,5 +190,101 @@ public function test_does_not_detect_uid_or_name_without_namespace(): void
         $this->assertEquals(ResourceInfoFactory::emptyResource(), $resource);
 
         @unlink($mountFile);
+        @unlink($namespaceFile);
+    }
+
+    public function test_returns_empty_resource_when_namespace_file_missing(): void
+    {
+        $namespaceFile = sys_get_temp_dir() . DIRECTORY_SEPARATOR . uniqid('namespace_missing_');
+        $mountFile = sys_get_temp_dir() . DIRECTORY_SEPARATOR . uniqid('mountinfo_missing_');
+        @unlink($namespaceFile);
+        @unlink($mountFile);
+        $resource = (new K8s($namespaceFile, $mountFile))->getResource();
+        $this->assertEquals(ResourceInfoFactory::emptyResource()->getAttributes(), $resource->getAttributes());
+    }
+
+    public function test_returns_empty_resource_when_namespace_file_unreadable(): void
+    {
+        $namespaceFile = sys_get_temp_dir() . DIRECTORY_SEPARATOR . uniqid('namespace_unreadable_');
+        file_put_contents($namespaceFile, 'test');
+        chmod($namespaceFile, 0000);
+        $mountFile = sys_get_temp_dir() . DIRECTORY_SEPARATOR . uniqid('mountinfo_unreadable_');
+        $resource = (new K8s($namespaceFile, $mountFile))->getResource();
+        $this->assertEquals(ResourceInfoFactory::emptyResource(), $resource);
+        chmod($namespaceFile, 0644);
+        unlink($namespaceFile);
+    }
+
+    public function test_returns_empty_namespace_when_namespace_file_empty(): void
+    {
+        $namespaceFile = sys_get_temp_dir() . DIRECTORY_SEPARATOR . uniqid('namespace_empty_');
+        file_put_contents($namespaceFile, '');
+        $mountFile = sys_get_temp_dir() . DIRECTORY_SEPARATOR . uniqid('mountinfo_empty_');
+        $resource = (new K8s($namespaceFile, $mountFile))->getResource();
+        $this->assertEquals('', $resource->getAttributes()->toArray()['k8s.namespace.name'] ?? 'null');
```

Comment:
it looks like the null default is not needed since it's equality checking?

Reply:
Yes